Base/etc/SystemServer.ini

```diff
@@ -20,6 +20,11 @@ Priority=low
 KeepAlive=1
 User=lookup
 
+[DHCPClient]
+Priority=high
+KeepAlive=1
+User=root
```

Comment:
* Why should it be high-priority? I'd rather wait a few more milliseconds to receive an IP address than have my WindowServer preempted by some networking stuff.
* I wish SystemServer could also lazy-spawn services on IP socket activity...
* Does it have to run as root?

Reply:
- I thought leaving the system disconnected for a while would be annoying, but honestly...the dhcp server gives the same IP we start with anyway, so I guess it doesn't matter.
- Theoretically, it should be spawned on interface changes, but I didn't find a way to do that.
- It does, it listens on port 67

Reply:
> I thought leaving the system disconnected for a while would be annoying

Priorities work on a scale of milliseconds — you can pretty much assume that any runnable thread, even if it has the lowest priority, will get a timeslice during the next few seconds, so I doubt it'll be noticable for a network configuration thingie. On the contrary, it matters a lot that the WindowServer and AudioServer are responsive under heavy system load, i.e. they can always react in milliseconds time, otherwise it feels that the system is "lagging".

Any kind of background activity that's not user-initiated nor visible (or audible) to the user should get a lower priority, meaning the kernel would rather delay the background activity than a realtime-y server in case of heavy system load. Also note that having a low priority doesn't really negativity impact you if the system is *not* under heavy load, i.e. if there's plenty of time to run all the runnable threads (the kernel will still sort runnable threads by their priority, so there is a small negative impact). In a way, you can think of a thread's priority as of how much you would like it to preempt other threads under load vs how much it should be preempted by others. I'd rather have WindowServer and AudioServer preempt DHCPClient 🙂

> It does, it listens on udp port 68

Ideally, SystemServer would support socket takeover for IP family sockets, then you'd be able to get the socket from it without binding it yourself.

Reply:
Fair point.
I had noticed a few other minor issues, I'll swap the priority along with them.

> SystemServer 

That would certainly make this look much better, we can give it a new user.
It would also play nicely with the next logical step of allowing the user to request a lease renewal.